test/display_test.py

```diff
@@ -782,6 +783,75 @@ def test_set_gamma_ramp(self):
 
         pygame.display.quit()
 
+class FullscreenToggleTests(unittest.TestCase):
+    __tags__ = [ "interactive" ]
+
+    screen = None
+    font = None
+    isfullscreen = False
+    aborted = False
+
+    WIDTH = 800
+    HEIGHT = 600
+
+    def setUp(self):
+        if self.screen is None:
+            pygame.init()
+            if sys.platform == "win32":
+                # known issue with windows, must have mode from pygame.display.list_modes()
+                # or window created with flag pygame.SCALED
+                self.screen = pygame.display.set_mode((self.WIDTH, self.HEIGHT), flags=pygame.SCALED)
+            else:
+                self.screen = pygame.display.set_mode((self.WIDTH, self.HEIGHT))
+            pygame.display.set_caption("Fullscreen Tests")
+            self.screen.fill((255, 255, 255))
+            pygame.display.flip()
+            self.font = pygame.font.Font(None, 32)
+
+    def quit(self):
+        if self.screen is not None:
+            if self.isfullscreen:
+                pygame.display.toggle_fullscreen()
+            pygame.quit()
+        self.aborted = True
```

Comment:
Maybe this method should be named `tearDown`? So that it does cleanup of calling `pygame.quit()` even after a successful run. Also storing `self.aborted` state is kinda redundant IMO, can probably be removed.

Reply:
I've taken on board what you said and have make changes accordingly